docs/Getting Started Part 2/index.md

```diff
@@ -0,0 +1,325 @@
+
+Once the components are registered, the HMI must notify sdl that it is ready to begin further communication using the [BasicCommunication.OnReady](../basiccommunication/onready) notification.
+
+Upon receipt of the OnReady notification, SDL will begin checking the availability of the different HMI components via a chain of requests:
+
+  * `UI.IsReady` - The display availability
+  * `VR.IsReady` - The voice recognition module availability
+  * `TTS.IsReady` - The Text-To-Speech module availability
+  * `Navigation.IsReady` - Navigation engine availability
+  * `VehicleInfo.IsReady` - Indicates whether vehicle information can be collected and provided
+  * `RC.IsReady` - Indicates whether vehicle RC modules are present and ready to communicate with SDL
+
+!!! note
+
+In the case of a WebSocket connection, the RPCs to each of the components is sent within each separate WebSocket connection.
+
+!!!
+
+|||
+IsReady Sequence
+![IsReady Sequence](./assets/IsReadySequence.png)
+|||
+
+!!! info
+
+If the response to any of the component `IsReady` requests contains `{"available": false}`, SDL will no longer communicate with that component.
+
+!!!
+
+ 
+## Registering for Notifications
+The HMI must also register for notifications individually using the following RPC format
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": -1,
+  "method": "MB.subscribeTo",
+  "params": {
+    "propertyName": notificationName
+  }
+}
+```
+
+"propertyName" is the name of the notification the HMI will receive from Core. Some examples include:
+
+  * Buttons.OnButtonSubscription
+  * BasicCommunication.OnAppRegistered
+  * BasicCommunication.OnAppUnregistered
+  * Navigation.OnVideoDataStreaming
+  * SDL.OnStatusUpdate
+
+Core's message broker will not route notifications to the HMI unless the notifications are subscribed to.
+
+!!! must
+
+The HMI must:
+  * Register its components
+  * Send the OnReady notification
+  * Respond to each of the `IsReady` RPCs
+  * Register for the notifications it would like to receive
+
+The above steps should only occur once per life cycle of SDL Core
+
+!!!
+
+## Communicating with SDL Core
+
+This section describes the message structure for communication between your HMI and SDL. The JSON RPC 2.0 format is taken as a basis.
+
+From this point forward the actors for exchanging messages will be considered:
+  - **Client** - can send requests and notifications
+  - **Server** - can provide responses to requests from a Client and send notifications
+
+## Request
+An RPC call is represented by sending a Request object to a Server. The Request object has the following properties
+
+| Property | Description    |
+| :------------- | :------------- |
+| "id"       | An identifier established by the Client. This value must be of unsigned int type in the frames of communication between your HMI and SDL. The value should never be Null. If "id" is not included the message is assumed to be a notification and the receiver should not respond.|
```

Comment:
```suggestion
| "id"       | An identifier established by the Client. This value must be of unsigned int type in the frames of communication between your HMI and SDL. The value should never be zero. If "id" is not included the message is assumed to be a notification and the receiver should not respond.|
```
null of unsigned int type is zero, makes more sense i think, at least shouldn't be null with capital N, $null_id used other places too, use the same value in other places with $null_id